inc/fasta/fasta.hpp

```diff
@@ -34,8 +34,28 @@ namespace ebi
       class IFasta
       {
       public:
+        /**
+         * Extract n base pairs from the contig starting from an offset
+         * @param contig - the name of the contig to extract the sequence from
+         * @param pos - the starting offset position
+         * @param length - the number of base pairs to be extracted
+         * @return the sequence string, empty if nothing can be extracted.
+         */
         virtual std::string sequence(const std::string& contig, const size_t pos, const size_t length) = 0;
-        virtual size_t count(const std::string& contig) const = 0;
+
+        /**
+         * Get the occurrences of sequences for a contig
```

Comment:
I think this may be a bit clearer:

```suggestion
         * Get the number of times a contig appears in a FASTA.
```